docs/buildkit/tutorials.md

```diff
@@ -0,0 +1,151 @@
+# Buildkit tutorials
+
+## Your First Pull-Request
+
+(TODO: Write a real tutorial!)
+
+Suppose we've just downloaded buildkit and want to prepare a patch for
+CiviCRM 4.7 (`master).  This requires downloading and installing the code for
+CiviCRM+CMS as well as writing and publishing the patch.
+
+```bash
+## Download Drupal 7.x and CiviCRM master
+civicrm-buildkit$ civibuild create drupal-demo --civi-ver master --url http://localhost:8001
+
+## Create a "fork" of civicrm-core on github.com for publishing changes
+civicrm-buildkit$ cd build/drupal-demo/sites/all/modules/civicrm
+civicrm$ hub fork
+
+## Create some changes locally
+civicrm$ git checkout origin/master -b master-mypatch
+civicrm$ vi CRM/Utils/Array.php
+civicrm$ civilint CRM/Utils/Array.php
+## Repeat vi/civilint until clean
+civicrm$ git commit CRM/Utils/Array.php
+
+## Publish our changes on github.com
+civicrm$ git push mygithubuser master-mypatch
+civicrm$ hub pull-request -b master
+
+## Make further changes based on feedback
+civicrm$ vi CRM/Utils/Array.php
+civicrm$ civilint CRM/Utils/Array.php
+civicrm$ git commit CRM/Utils/Array.php
+civicrm$ git push mygithubuser
+```
+
+Please note:
+ * This example targets CiviCRM v4.7 (`master`). If you're targetting an older version,
+   change all references from `master` to appropriate branch (e.g. `4.6` or `4.5` or `4.4`).
+ * A build may include several different git repositories. The
+   commands should look about the same on any repository, although different
+   git repositories may use different names for their versions/branches (eg the
+   civicrm-core, civicrm-joomla, and civicrm-wordpress repositories have
+   branches named `master`, but the civicrm-drupal repository has a branch named
+   `7.x-master`).
+
+## Housekeeping
+
+(TODO: Write a real tutorial!)
+
+From time-to-time, you may want to update your code. It's also a good idea
+to double-check that your git repos have checked out the normal/expected
+branches.
+
+```bash
+civicrm-buildkit$ cd build/drupal-demo
+
+drupal-demo$ git scan status
+## If any repos look unclean, figure out why... and clean them up!
+
+## Then:
+drupal-demo$ git scan update
+
+## If you think the DB or config structure has changed, then optionally:
+drupal-demo$ civibuild reinstall drupal-demo
+```
+
+## Unit Tests
+
+```bash
+civicrm-buildkit$ cd build/drupal-demo/sites/all/modules/civicrm/tools
+
+## Run all the API tests
+tools$ ./scripts/phpunit api_v3_AllTests
```

Comment:
Not a blocker... but now that 4.7 is over half the install-base, it's tempting to switch this over to vanilla phpunit notation, e.g.

```php
env CIVICRM_UF phpunit4 tests/phpunit/api/v3/AllTests.php
```

It is a few characters longer, but it lays better foundation for when folks move on to writing new test suites (outside `civicrm-core`) or executing tests in other ways (like PHPStorm or Jenkins).